openquake/mbt/tools/fault_modeler/fault_source_modeler.py

```diff
@@ -0,0 +1,361 @@
+#
+# Copyright (C) 2015-2018 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors: Julio Garcia, Richard Styron, Valerio Poggi
+# Last modify: 10/09/2018
+
+# -----------------------------------------------------------------------------
+
+import sys
+import ast
+import json
+from copy import deepcopy
+
+import configparser
+import warnings
+
+import openquake.mbt.tools.fault_modeler.fault_modeling_utils as fmu
+from openquake.hazardlib.sourcewriter import write_source_model
+from openquake.baselib import sap
+
+# -----------------------------------------------------------------------------
+
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+# Parameters required from the fault modeler
+option_types = {'b_value': float,
+                'M_min': float,
+                'M_max': float,
+                'bin_width': float,
+                'aseismic_coefficient': float,
+                'rupture_aspect_ratio': float,
+                'rupture_mesh_spacing': float,
+                'minimum_fault_length': float,
+                'tectonic_region_type': str,
+                'upper_seismogenic_depth': float,
+                'lower_seismogenic_depth': float,
+                'magnitude_scaling_relation': str,
+                'width_scaling_relation': str}
+
+
+# -----------------------------------------------------------------------------
+
+def build_fault_model(cfg_file=None,
+                      geojson_file=None,
+                      xml_output=None,
+                      black_list=None,
+                      select_list=None,
+                      project_name=None,
+                      width_method='seismo_depth',
+                      oqt_source=False,
+                      param_map=None,
+                      defaults=None,
+                      **kwargs):
+    """
+    Main interface to create the fault source model from an active fault
+    database in geojson format.
+    Priority for the optional parameters is set to:
+        1) single arguments
+        2) dictionary
+        3) .ini file
+    """
+
+    param_map_local = deepcopy(fmu.param_map)
+    defaults_local = deepcopy(fmu.defaults)
+
+    # Import arguments from INI configuration file
+    if cfg_file is not None:
+        cfg_dict = read_config_file(cfg_file)
+
+        if 'config' in cfg_dict:
```

Comment:
Please replace all ``cfg_dict['config']``of this block with ``config`` where config is:
```python
    config = cfg_dict['config']
```
Placed here.